### PR TITLE
Specify node type so that the log filename can get resolved

### LINKED ISSUE #15967: Log files are stored as ${sys:druid.node.type}.log
I am using a k8s based deployment of Druid and see that each node creates log files that are named as 
```${sys:druid.node.type}.log```

The variable is not getting resolved to the right node type. The log files are stored as ```${sys:druid.node.type}.log``` or ```${sys:druid.node.type}-<date>.log```

### Affected Version

Druid version: 28.0.0

### Description

Due to the above problem, I am not able to index the logs appropriately. Am I missing something here?

### PR DESCRIPTION
**Pull request by fectrain**
Fixes #15967 

### Description

Specify node type in `druid.sh` so that the log filename(defined in log4j2.xml) can get resolved

image

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: distribution/docker/druid.sh [hunk 1]
```diff
@@ -172,6 +172,9 @@ if [ -n "$DRUID_MAXDIRECTMEMORYSIZE" ]; then setJavaKey ${SERVICE} -XX:MaxDirect
 # However this behavior is not part of the spec and is thus implementation specific
 JAVA_OPTS="$(cat $SERVICE_CONF_DIR/jvm.config | xargs) $JAVA_OPTS"
 
+# specify node type used for log4j2.xml
```
Comment (asdf2014) [inline distribution/docker/druid.sh:175]:
```suggestion
# Specify node type used for log4j2.xml
```

**Review by asdf2014 — APPROVED** [review]

**Comment by cryptoe** [comment]
@fectrain Could you share a screen shot of the state before this patch as well ?

**Comment by fectrain** [comment]
> @fectrain Could you share a screen shot of the state before this patch as well ?

Hi, can just refer #15967 , which is just look likes `${sys:druid.node.type}.log`

**Comment by Subhashini2610** [comment]
@fectrain @asdf2014 Thanks for this fix!!! Please let me know in which Druid release would this be present? I somehow do not see it in 30.0.0

**Pull request merged**